src/plugins/discover/public/application/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState } from 'react';
+import { sortBy } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { UiStatsMetricType } from '@kbn/analytics';
+import {
+  EuiTitle,
+  EuiHideFor,
+  EuiShowFor,
+  EuiButton,
+  EuiBadge,
+  EuiFlyoutHeader,
+  EuiFlyout,
+  EuiSpacer,
+  EuiIcon,
+  EuiLink,
+  EuiPortal,
+} from '@elastic/eui';
+import { DiscoverIndexPattern } from './discover_index_pattern';
+import { IndexPatternAttributes } from '../../../../../data/common';
+import { SavedObject } from '../../../../../../core/types';
+import { IndexPatternField, IndexPattern } from '../../../../../data/public';
+import { getDefaultFieldFilter } from './lib/field_filter';
+import { DiscoverSidebar } from './discover_sidebar';
+import { DiscoverServices } from '../../../build_services';
+
+export interface DiscoverSidebarResponsiveProps {
+  /**
+   * Determines whether add/remove buttons are displayed non only when focused
+   */
+  alwaysShowActionButtons?: boolean;
+  /**
+   * the selected columns displayed in the doc table in discover
+   */
+  columns: string[];
+  /**
+   * a statistics of the distribution of fields in the given hits
+   */
+  fieldCounts: Record<string, number>;
+  /**
+   * hits fetched from ES, displayed in the doc table
+   */
+  hits: Array<Record<string, unknown>>;
```

Comment:
nit: Could this be `ElasticSearchHit[]`?

Reply:
of course it could, and now it is! thx!